src/components/InputSearch/InputSearch.spec.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import { render, fireEvent } from '@testing-library/react';
+
+import InputSearch from './InputSearch';
+
+// Mock das props
+const mockProps = {
+  name: 'testInput',
+  type: 'text',
+  value: '',
+  required: false,
+  placeholder: 'Test Placeholder',
+  errorMessage: 'Test Error Message',
+  readonly: false,
+  errors: false,
+  onFocus: jest.fn(),
+  handleClear: jest.fn(),
+  onChange: jest.fn(),
+};
+
+test('renders InputSearch component', () => {
+  const { getByLabelText, getByPlaceholderText } = render(
+    <InputSearch {...mockProps} />
+  );
+
+  const inputElement = getByPlaceholderText('Test Placeholder');
+  const labelElement = getByLabelText('Test Placeholder');
+
+  expect(inputElement).toBeInTheDocument();
+  expect(labelElement).toBeInTheDocument();
+});
+
+test('handles input value correctly', () => {
+  const { getByPlaceholderText } = render(<InputSearch {...mockProps} />);
+  const inputElement = getByPlaceholderText(
+    'Test Placeholder'
+  ) as HTMLInputElement;
+
+  fireEvent.change(inputElement, { target: { value: 'Test Value' } });
```

Comment:
o `fireEvent` é uma versão depreciada/mais básica do `userEvent`, então tenta usa-lo sempre. Apenas troque pro `fireEvent` em casos muitos específicos (testar drag and drop por exemplo)

Links úteis: 
https://cursos.alura.com.br/forum/topico-podemos-usar-o-userevent-ao-inves-do-fireevent-310309
https://testing-library.com/docs/guide-events/